homeassistant/components/mqtt/sensor.py

```diff
@@ -256,13 +272,17 @@ def _update_state(msg):
                     self._expiration_trigger()
 
                 # Set new trigger
-                expiration_at = dt_util.utcnow() + timedelta(seconds=expire_after)
+                expiration_at: datetime = dt_util.utcnow() + timedelta(
+                    seconds=expire_after
+                )
 
                 self._expiration_trigger = async_track_point_in_utc_time(
                     self.hass, self._value_is_expired, expiration_at
                 )
 
+            payload: ReceivePayloadType | date | datetime | None
             payload = self._template(msg.payload, default=self.native_value)
+            assert not isinstance(payload, bytes)
```

Comment:
I think there is an inconsistency here with `async_render_with_possible_json_value`
The return_type doesn't match the default value passed in.

Reply:
The `native_value` could supply some additional types.

Reply:
Yes, but that means `async_render_with_possible_json_value` signature is wrong :(

Reply:
I know, may be an extra method could help, other parameters are not used frequently.

Reply:
Solve using a const instead of the native value